Ginger/Ginger/UserControlsLib/ActionInputValueUserControlLib/ActionInputValueUserControl.xaml.cs

```diff
@@ -36,17 +40,136 @@ public partial class ActionInputValueUserControl : UserControl
+        bool mBrowseNeeded = false;
+        Environment.SpecialFolder mFolderType = Environment.SpecialFolder.MyComputer;
+        Ginger.Activities.UCValueExpression.eBrowserType mBrowseType = Ginger.Activities.UCValueExpression.eBrowserType.File;
+        string mFileType = string.Empty;
+        public ActionInputValueUserControl(Context context, ActInputValue actInputValue, List<Attribute> actionParamProperties)
         {
             InitializeComponent();
 
             mActInputValue = actInputValue;
             mContext = context;
+            mActionParamProperties = actionParamProperties;
 
             ResetControls();
 
+            mLabel = char.ToUpper(mActInputValue.Param[0]) +  mActInputValue.Param.Substring(1);   // capitilize first letter only of the param name
+
+            SetParamLayout();
             SetControlToInputValue();
+            SetParamValidations();
+        }
+
+        private void SetParamValidations()
+        {
+            if (mActionParamProperties == null)
+            {
+                return;
+            }
+
+            foreach (Attribute attr in mActionParamProperties)
+            {             
+                if (attr.GetType() == typeof(MandatoryAttribute))
+                {
+                    xTextBoxInputPnl.Background = Brushes.Orange;
+                    //TODO: add AddValidationRule in the UC - but there are 3 UCs !? !!!!!!!!!!!!!!!                  
+                    xTextBoxInputTextBox.ValueTextBox.AddValidationRule(new EmptyValidationRule());                    
+                }
+                else if(attr.GetType() == typeof(MaxValueAttribute))
+                {                           
+                    xTextBoxInputTextBox.ValueTextBox.AddValidationRule(new MaxValueValidationRule(((MaxValueAttribute)attr).MaxValue));                    
+                }
+                else if(attr.GetType() == typeof(MinValueAttribute))
+                {                    
+                    xTextBoxInputTextBox.ValueTextBox.AddValidationRule(new MinValueValidationRule(((MinValueAttribute)attr).MinValue));                                        
+                }          
+                else if(attr.GetType() == typeof(MaxLengthAttribute))
+                {
+                    xTextBoxInputTextBox.ValueTextBox.AddValidationRule(new MaxLengthValidationRule(((MaxLengthAttribute)attr).MaxLength));
+                }
+                else if(attr.GetType() == typeof(MinLengthAttribute))
+                {
+                    xTextBoxInputTextBox.ValueTextBox.AddValidationRule(new MinLegthValidationRule(((MinLengthAttribute)attr).MinLength));
+                }
+                else if(attr.GetType() == typeof(InvalidValueAttribute))
+                {
+                    InvalidValueValidationRule IVR = (InvalidValueValidationRule)xTextBoxInputTextBox.ValueTextBox.GetValidationRule(TextBox.TextProperty, typeof(InvalidValueValidationRule));                   
```

Comment:
Change IVR to InvalidValueValidationRule  - using var naming as suggested by intellisense